src/node_url.cc

```diff
@@ -75,6 +75,73 @@ void BindingData::Deserialize(v8::Local<v8::Context> context,
   CHECK_NOT_NULL(binding);
 }
 
+std::string EncodePathChars(const std::string& input_str, bool windows) {
```

Comment:
Non-blocking nit: Really not a big fan of bool arguments to methods like this. Would prefer an enum.

Reply:
PS: If you use a template argument you can make isWindows checks a constexpr and eliminate the branch.